### ISSUE TITLE
arrow-parens + async arrow function bug

### ISSUE THREAD

**Issue by phaux** [eslint/eslint#6311]
```js
setTimeout(async () => {}, 1)
```

That causes an error:

>Expected parentheses around arrow function argument. at line 1 col 12

ESLint 2.11.1 using Babel-ESLint parser

**Comment by eslintbot** [comment]
Hi @phaux, thanks for the issue. It looks like there's not enough information for us to know how to help you. If you're reporting a bug, please be sure to include:

1. The version of ESLint you are using (run `eslint -v`)
2. What you did (the source code and ESLint configuration)
3. The actual ESLint output complete with numbers
4. What you expected to happen instead

Requesting a new rule? Please see [Proposing a New Rule](http://eslint.org/docs/developer-guide/contributing/new-rules) for instructions.

If it's something else, please just provide as much additional information as possible. Thanks!

**Comment by vitorbal** [comment]
Could this be related to #6274 or #6281 ?
I'm guessing this is probably a babel-eslint issue with `async` and the `arrow-parens` rule. @hzoo does it ring a bell?

**Comment by hzoo** [comment]
Yep, I would use https://github.com/babel/eslint-plugin-babel#rules - `babel/arrow-parens`. It sounds like a "babel-eslint" issue but its just that the rule can't handle newer syntax so currently we have to actually rewrite it as a separate rule + another if check.

**Comment by vitorbal** [comment]
@hzoo thanks, that was a fast reply :-)
@phaux Could you try with `babel/arrow-parens` and let us know if that works out for you?

**Comment by phaux** [comment]
I didn't know about eslint-plugin-babel. I will try it out. Thanks
